src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -636,8 +649,6 @@ private Result calculateScoreForResult(Set<ProgrammingExerciseTestCase> testCase
         else if (!testCases.isEmpty() && !result.getFeedbacks().isEmpty() && !testCaseFeedback.isEmpty()) {
             addFeedbackTestsNotExecuted(result, exercise, staticCodeAnalysisFeedback);
         }
-        // Case 3: If there is no test case feedback, the build has failed, or it has previously fallen under case 2. In this case we just return the original result without
-        // changing it.
```

Comment:
I think that comment was helpful to describe which other cases are possible as that is not obvious from the if/elseIf conditions above. Can you add it back?

Reply:
Will do